spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -305,61 +342,74 @@ public Statistics estimateStatistics() {
   @Override
   public String toString() {
     return String.format(
-        "IcebergScan(table=%s, type=%s, filters=%s, caseSensitive=%s)",
-        table, lazySchema().asStruct(), filterExpressions, caseSensitive);
+        "IcebergScan(table=%s, type=%s, filters=%s, caseSensitive=%s, batchedReads=%s)",
+        table, lazySchema().asStruct(), filterExpressions, caseSensitive, enableBatchRead);
   }
 
-  private static class ReadTask implements InputPartition<InternalRow>, Serializable {
-    private final CombinedScanTask task;
+  @SuppressWarnings("checkstyle:VisibilityModifier")
+  private static class BaseReadTask implements Serializable {
+    final CombinedScanTask task;
     private final String tableSchemaString;
     private final String expectedSchemaString;
-    private final Broadcast<FileIO> io;
-    private final Broadcast<EncryptionManager> encryptionManager;
-    private final boolean caseSensitive;
-    private final boolean localityPreferred;
+    final Broadcast<FileIO> io;
+    final Broadcast<EncryptionManager> encryptionManager;
+    final boolean caseSensitive;
 
     private transient Schema tableSchema = null;
     private transient Schema expectedSchema = null;
-    private transient String[] preferredLocations;
 
-    private ReadTask(CombinedScanTask task, String tableSchemaString, String expectedSchemaString,
-                     Broadcast<FileIO> io, Broadcast<EncryptionManager> encryptionManager,
-                     boolean caseSensitive, boolean localityPreferred) {
+    BaseReadTask(CombinedScanTask task, String tableSchemaString, String expectedSchemaString,
+        Broadcast<FileIO> io, Broadcast<EncryptionManager> encryptionManager,
```

Comment:
Style: We typically indent method args from the same location. That would be either like it was before, or moving all of the method args to use the 4-space continuation indent.